Source/Provers/SMTLib/SMTLibProcess.cs

```diff
@@ -68,6 +69,17 @@ public SMTLibProcess(SMTLibOptions libOptions, SMTLibProverOptions options)
       }
     }
 
+    private void prover_Exited(object sender, EventArgs e)
+    {
+      lock (this) {
+        if (outputReceivers.TryDequeue(out var source)) {
+          source.SetResult(null);
+        }
```

Comment:
There's a while so it should remove all of them

Reply:
There cannot be two things waiting for it in our current code be because we always prefix the calls to `getProverResponse` with `await`. However, as a safeguard for future problems that are unlikely to happend, we added the comment that this class is not thread-safe, meaning it should be also used only with one thread (which is what happens, it is owned by a Checker that runs the prover on only one thread.)

Reply:
> I guess you reviewed an older version with just an if :)

!!  Precisely right :)